master/pkg/tasks/task.go

```diff
@@ -225,7 +225,8 @@ func TrialEnvVars(t TaskSpec, rendezvousPorts []string) map[string]string {
 	envVars["DET_EXPERIMENT_CONFIG"] = jsonify(exp.ExperimentConfig)
 	envVars["DET_HPARAMS"] = jsonify(exp.HParams)
 	envVars["DET_INITIAL_WORKLOAD"] = jsonify(exp.InitialWorkload)
-	envVars["DET_LATEST_CHECKPOINT"] = jsonify(exp.LatestCheckpoint)
+
+	envVars["DET_LATEST_CHECKPOINT"] = "/run/determined/checkpoint.json"
```

Comment:
I think if we put this file directly in this directory, it would be the only file there (everything else is organized into some directory.  What if we put it in a directory called "/run/determined/train" or something?  Then we could also put the `entrypoint.sh` there.

Reply:
Agreed with you on moving the checkpoint file to a new /run/determined/train directory. For consistency purposes, it seems that for the other entrypoint files (gc-checkpoints, notebook-entrypoint, tensorboard-entrypoint, etc.) remain in the workdir, so I’m thinking this entrypoint file fits in there as well?

Reply:
I don't see any value in "consistency" when the current way is broken.  It's fine to put files in the working directory for tensorboard or gc-checkpoints, because those are not interactive commands.  However, notebooks, training, and shells are interactive, and we give the user a `--context` option on the cli, and then we promise them that they will have that context directory as their working directory.  Then we inject our own scripts into that context directory, but there's no real reason why.

FWIW, the shell entrypoint does not get put into the working directory, so it's not like this would be the first entrypoint to break the original pattern.